src/query/block/container.go

```diff
@@ -312,40 +341,31 @@ func (it *containerSeriesIter) Current() Series {
 	return it.its[it.idx].Current()
 }
 
-func (it *containerSeriesIter) Meta() Metadata {
-	// NB: metadata should be identical for each series in the contained block.
-	if len(it.its) == 0 {
-		return Metadata{}
-	}
-
-	return it.its[0].Meta()
-}
-
-// Unconsolidated returns the unconsolidated version for the block
 func (b *containerBlock) Unconsolidated() (UnconsolidatedBlock, error) {
 	if b.err != nil {
 		return nil, b.err
 	}
 
-	ucBlock := &ucContainerBlock{
-		blocks: make([]UnconsolidatedBlock, 0, len(b.blocks)),
-	}
-
-	for i, bl := range b.blocks {
+	blocks := make([]UnconsolidatedBlock, 0, len(b.blocks))
+	for _, bl := range b.blocks {
 		unconsolidated, err := bl.Unconsolidated()
 		if err != nil {
 			b.err = err
 			return nil, err
 		}
 
-		ucBlock.blocks[i] = unconsolidated
+		blocks = append(blocks, unconsolidated)
```

Comment:
Nice

Reply:
As I mentioned prior, yeah maybe we should just be using `make(thing, 0, expectedLength)` and `append(...)` everywhere possible just to try to mainly avoid this being a thing in other places.